src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -440,6 +487,29 @@ private String getInlineSummary() {
                 .replaceAll("");
     }
 
+    /**
+     * Checks the inline return for forbidden fragments.
+     *
+     * @param ast javadoc root node.
+     */
+    private void validateInlineReturnTag(DetailNode ast) {
+        // the presence of the tag is checked by `containsInlineReturnTag`
+        final DetailNode inlineTagNode = getInlineTagNode(ast).orElseThrow();
```

Comment:
`getInlineTagNode(ast).get();`

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html#get()
I know that in jdk javadoc it is recommended to use `orElseThrow` but we do not allow Checks to throw exceptions. I know it is the same as operation on None in Optional, but it make code is a bit weird to read and we care about readability a lot.

Reply:
I can't do that or [the IDEA inspections check](https://teamcity.jetbrains.com/viewLog.html?buildId=3796456&tab=Inspection&buildTypeId=Checkstyle_IdeaInspectionsPullRequest) fails. Would you prefer I suppress that inspection? Or do you have another idea?

Reply:
```
src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc
SummaryJavadocCheck.java (1)
[496](https://teamcity.jetbrains.com/viewLog.html?buildId=3796456&tab=Inspection&buildTypeId=Checkstyle_IdeaInspectionsPullRequest#): validateInlineReturnTag() Optional.get() without 'isPresent()' 
check
```

Reply:
yes, this is channlenge .....
@pbludov , can you help, I will be off for a day.

Reply:
I pushed a change, @octylFractal , please squash all in one commit if you agree.

Reply:
I made a slight modification, using `flatMap` to have only one `orElse`. That is committed now.